compiler/varpartitions.nim

```diff
@@ -578,24 +578,25 @@ proc borrowingAsgn(c: var Partitions; dest, src: PNode) =
   if dest.kind == nkSym:
     if directViewType(dest.typ) != noView:
       borrowFrom(c, dest.sym, src)
-  elif dest.kind in {nkHiddenDeref, nkDerefExpr, nkBracketExpr}:
-    case directViewType(dest[0].typ)
```

Comment:
The culprit: you need to query the symbol for its type in the definition instead of the assignment. Otherwise it cannot preserve the view type infos.